src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java

```diff
@@ -92,6 +106,12 @@ private boolean takeTheDirectPath(@NotNull BlockPos target, int range, boolean o
      */
     public boolean walkToBlock(@NotNull BlockPos target, int range, boolean onMove)
     {
+        if(!target.equals(this.target))
+        {
+            reset();
+            this.target = target;
+        }
+
         final double distanceToPath = worker.getColonyJob() instanceof JobBuilder
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) '?' should be on a new line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AOperator_Wrap)